cli/src/cmd/cast/run.rs

```diff
@@ -0,0 +1,164 @@
+use crate::{
+    cmd::{forge::build::BuildArgs, Cmd},
+    opts::evm::EvmArgs,
+    utils,
+};
+
+use ansi_term::Colour;
+use clap::Parser;
+use ethers::types::{Address, Bytes, U256};
+
+use forge::{
+    executor::{
+        opts::EvmOpts, DatabaseRef, DeployResult, Executor, ExecutorBuilder, RawCallResult,
+    },
+    trace::TraceKind,
+    CALLER,
+};
+use foundry_config::{figment::Figment, Config};
+use foundry_utils::{encode_args, IntoFunction};
+use hex::ToHex;
+
+// Loads project's figment and merges the build cli arguments into it
+foundry_config::impl_figment_convert!(RunArgs, opts, evm_opts);
+
+#[derive(Debug, Clone, Parser)]
+pub struct RunArgs {
+    #[clap(help = "the bytecode to execute")]
+    pub bytecode: String,
+
+    // Optional Calldata
+    #[clap(help = "the calldata to pass to the contract")]
+    pub calldata: Option<String>,
+
+    /// Open the bytecode execution in debug mode
+    #[clap(long, help = "debug the bytecode execution")]
+    pub debug: bool,
+
+    #[clap(flatten)]
+    opts: BuildArgs,
+
+    #[clap(flatten)]
+    pub evm_opts: EvmArgs,
+}
+
+impl Cmd for RunArgs {
+    type Output = ();
+
+    fn run(self) -> eyre::Result<Self::Output> {
+        // Load figment
+        let figment: Figment = From::from(&self);
+        let evm_opts = figment.extract::<EvmOpts>()?;
+        let verbosity = evm_opts.verbosity;
+        let config = Config::from_provider(figment).sanitized();
+
+        // Parse bytecode string
+        let bytecode_vec = self.bytecode.strip_prefix("0x").unwrap_or(&self.bytecode);
+        let parsed_bytecode = Bytes::from(hex::decode(bytecode_vec)?);
```

Comment:
I think this should work as well
```suggestion
        let bytecode_vec:Bytes = self.bytecode.parse()?;
```

Reply:
done, thanks :)